### ISSUE TITLE
Does lein-ring support reload changed namespaces somehow?

### ISSUE THREAD

**Issue by ertugrulcetin** [weavejester/lein-ring#187]
When I add a new namespace to my project or change couple of those my ring application throws an exception that says namespace couldn't be found.I can see changes when I don't touch the namespaces which is very cool but I also want to see the changes when I change (add/delete/update) namespaces.

Does lein-ring support this built in with some configuration?

**Comment by weavejester** [comment]
Adding namespaces work the same way; it's likely that you're just trying to access the namespace without an associated `require`.

**Comment by ertugrulcetin** [comment]
Yeah I need to restart lein app when I introduce new namespace or change one of them, is there any chance to do some adjustment and make this namespace loading automatically?

**Comment by weavejester** [comment]
It should automatically load by default. Could you give some more information about this issue? So for instance:

* What does the error say?
* What is the stacktrace?
* What does your `ns` declaration look like of both the file being loaded, and the file doing the loading?
* What are the filenames (perhaps you've added a `-` instead of an `_`)?
* Is your project open source, or can you create a project that demonstrates the problems you're having?
* What version of Lein-Ring are you using?
* What does your `lein deps :tree` output look like?

**Comment by ertugrulcetin** [comment]
I was using "0.11.0" but now updated to "0.12.0" and it works I could not reproduce now.
I'm going to try to reproduce , if it happens again I will provide all information you need.Thank you so much for the quick response.

**Comment by ertugrulcetin** [comment]
@weavejester I think the problem is Clojure.I'm getting Compiler exception:
`Alias auth already exists`

reproduce steps:
first I have this ->` [cl.controller.auth :as auth] `
after I change this to that-> ` [cl.controller.auth-service :as auth] `

I guess Clojure caches the first **cl.controller.auth** namespace's aliases which is auth. Then I refactor(rename) the **cl.controller.auth** to **cl.controller.auth-service** so I get that exception.

Is there any way to unalias the alias and redefine it?

This is the reason why I have to restart the app.

**Comment by MichaelBlume** [comment]
Personally I like to use the reload function from
https://github.com/clojure/tools.namespace but it's a very big hammer and
it's some work to be sure your code will cooperate with it

**Comment by ertugrulcetin** [comment]
@weavejester any updates regarding that [comment](https://github.com/weavejester/lein-ring/issues/187#issuecomment-309224778)?

**Comment by weavejester** [comment]
Updates in what respect? Lein-Ring uses Clojure's reload mechanism, which won't work in the case where you're trying to overwrite an existing namespace alias. It doesn't currently make use of `refresh` in tools.namespace, with might solve the issue. A workaround is to run [ns-unalias](https://clojure.github.io/clojure/clojure.core-api.html#clojure.core/ns-unalias) manually.

**Comment by ertugrulcetin** [comment]
got it thank you so much!
